python/dgl/graphbolt/impl/torch_based_feature_store.py

```diff
@@ -560,28 +564,41 @@ class TorchBasedFeatureStore(BasicFeatureStore):
     ...     gb.OnDiskFeatureData(domain="node", type="paper", name="feat",
     ...         format="numpy", path="/tmp/node_feat.npy", in_memory=False),
     ... ]
-    >>> feature_sotre = gb.TorchBasedFeatureStore(feat_data)
+    >>> feature_store = gb.TorchBasedFeatureStore(feat_data)
     """
 
-    def __init__(self, feat_data: List[OnDiskFeatureData]):
+    def __init__(
+        self,
+        feat_data: List[OnDiskFeatureData],
+        disk_based_feature_list: Set[FeatureKey] = None,
```

Comment:
why do you need this? If our new implementation beats mmap, why not just replace all mmap silently by respecting in_memory = false?

Reply:
How do I load ogbn-papers100M into our example with DiskBasedFeature? I have to find the downloaded yaml file and change the in_memory field.

Reply:
Even if I do, then mmap codepath is going to be impossible. DiskBasedFeature is available only on Linux and makes sense to use together with CPUCachedFeature. Mmap does not need CPUCachedFeature.

Reply:
The yaml can be changed while you load them. Don't ask user to specify what caching mechanism to choose, do it for them, check availability of CPUCachedFeature, otherwise fallback to mmap.

Reply:
How do I pick the capacity of the CPUCachedFeature?

Reply:
mmap does not require one to set how much memory to use as it is automatically managed by the OS kernel. Our approach requires us to pass it as argument.

Reply:
This parameter is needed to override the in_memory option in user's script. Normally, I don't want to expose it in TorchBasedFeatureStore but OnDiskDataset.load needs to somehow pass it to TorchBasedFeatureStore.